src/components/organisms/NavBarSchedule/NavBarSchedule.tsx

```diff
@@ -89,21 +97,66 @@ export const NavBarSchedule: React.FC<NavBarScheduleProps> = ({
 
   const [selectedDayIndex, setSelectedDayIndex] = useState(0);
 
+  const { minDate, maxDate } = relatedVenueEvents.reduce(
+    (resultObj, event) => {
+      if (resultObj.minDate === 0 || resultObj.maxDate === 0) {
+        resultObj.minDate =
+          resultObj.minDate === 0 ? event.start_utc_seconds : resultObj.minDate;
+
+        const secondsLength = minutesToSeconds(event.duration_minutes);
+        resultObj.maxDate =
+          resultObj.maxDate === 0
+            ? event.start_utc_seconds + secondsLength
+            : resultObj.maxDate;
+      }
+
+      if (event.start_utc_seconds < resultObj.minDate) {
+        resultObj.minDate = event.start_utc_seconds;
+      }
+
+      if (
+        event.start_utc_seconds + minutesToSeconds(event.duration_minutes) >
+        resultObj.maxDate
+      ) {
+        const secondsLength = minutesToSeconds(event.duration_minutes);
+
+        resultObj.maxDate = event.start_utc_seconds + secondsLength;
+      }
+
+      return resultObj;
+    },
+    { minDate: 0, maxDate: 0 }
+  );
+  const milisecondStart = new Date(secondsToMilliseconds(minDate)).getTime();
+  const milisecondFinish = new Date(secondsToMilliseconds(maxDate)).getTime();
+  const dayDifference = Math.round(
+    millisecondsToHours(milisecondFinish - milisecondStart) / 24
+  );
+
   const weekdays = useMemo(() => {
     const formatDayLabel = (day: Date | number) => {
       if (isScheduleTimeshifted) {
-        return format(day, "E, LLL d");
+        return format(day, "do");
       } else {
         return formatDateRelativeToNow(day, {
-          formatOtherDate: (dateOrTimestamp) => format(dateOrTimestamp, "E"),
+          formatOtherDate: (dateOrTimestamp) => format(dateOrTimestamp, "do"),
+          formatTomorrow: (dateOrTimestamp) => format(dateOrTimestamp, "do"),
         });
       }
     };
 
-    return range(SCHEDULE_SHOW_DAYS_AHEAD).map((dayIndex) => {
+    return range(dayDifference).map((dayIndex) => {
       const day = addDays(firstDayOfSchedule, dayIndex);
```

Comment:
I think `firstDayOfSchedule` should be calculated based on events information:  max(minDate, today)
and then we can get rid of `isScheduleTimeshifted` logic. Because we'll show only the event dates that have at least something.